x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/sections/alert_details/components/alert_details.test.tsx

```diff
@@ -0,0 +1,519 @@
+      .find('[name="mute"]')
+      .first();
+
+    expect(enableButton.props()).toMatchObject({
+      checked: false,
+      disabled: false,
+    });
+  });
+
+  it('should render an muted button when alert is muted', () => {
+    const alert = mockAlert({
+      enabled: true,
+      muteAll: true,
+    });
+
+    const alertType = {
+      id: '.noop',
+      name: 'No Op',
+    };
+
+    const enableButton = shallow(
+      <AlertDetails alert={alert} alertType={alertType} actionTypes={[]} {...mockAlertApis} />
+    )
+      .find(EuiSwitch)
+      .find('[name="mute"]')
+      .first();
+
+    expect(enableButton.props()).toMatchObject({
+      checked: true,
+      disabled: false,
+    });
+  });
+
+  it('should mute the alert when alert is unmuted and button is clicked', () => {
+    const alert = mockAlert({
+      enabled: true,
+      muteAll: false,
+    });
+
+    const alertType = {
+      id: '.noop',
+      name: 'No Op',
+    };
+
+    const muteAlert = jest.fn();
+    const enableButton = shallow(
+      <AlertDetails
+        alert={alert}
+        alertType={alertType}
+        actionTypes={[]}
+        {...mockAlertApis}
+        muteAlert={muteAlert}
+      />
+    )
+      .find(EuiSwitch)
+      .find('[name="mute"]')
+      .first();
+
+    enableButton.simulate('click');
+    const handler = enableButton.prop('onChange');
+    expect(typeof handler).toEqual('function');
+    expect(muteAlert).toHaveBeenCalledTimes(0);
+    handler!({} as React.FormEvent);
+    expect(muteAlert).toHaveBeenCalledTimes(1);
+  });
+
+  it('should unmute the alert when alert is muted and button is clicked', () => {
+    const alert = mockAlert({
+      enabled: true,
+      muteAll: true,
+    });
+
+    const alertType = {
+      id: '.noop',
+      name: 'No Op',
+    };
+
+    const unmuteAlert = jest.fn();
+    const enableButton = shallow(
+      <AlertDetails
+        alert={alert}
+        alertType={alertType}
+        actionTypes={[]}
+        {...mockAlertApis}
+        unmuteAlert={unmuteAlert}
+      />
+    )
+      .find(EuiSwitch)
+      .find('[name="mute"]')
+      .first();
+
+    enableButton.simulate('click');
+    const handler = enableButton.prop('onChange');
+    expect(typeof handler).toEqual('function');
+    expect(unmuteAlert).toHaveBeenCalledTimes(0);
+    handler!({} as React.FormEvent);
+    expect(unmuteAlert).toHaveBeenCalledTimes(1);
+  });
+
+  it('should disabled mute button when alert is disabled', () => {
+    const alert = mockAlert({
+      enabled: false,
+      muteAll: false,
+    });
+
+    const alertType = {
+      id: '.noop',
+      name: 'No Op',
+    };
+
+    const enableButton = shallow(
+      <AlertDetails alert={alert} alertType={alertType} actionTypes={[]} {...mockAlertApis} />
+    )
+      .find(EuiSwitch)
+      .find('[name="mute"]')
+      .first();
+
+    expect(enableButton.props()).toMatchObject({
+      checked: false,
+      disabled: true,
+    });
+  });
```

Comment:
This unit tests looks awesome! I definitely should follow your style

Reply:
<3